internals/overlord/state/identities_test.go

```diff
@@ -538,3 +538,43 @@ func (s *identitiesSuite) TestIdentities(c *C) {
 	identities["changed"] = &state.Identity{}
 	c.Assert(identities2, DeepEquals, expected)
 }
+
+func (s *identitiesSuite) TestIdentityFromInputs(c *C) {
```

Comment:
Maybe also add a test case what happens when `st.IdentityFromInputs()` is called without `st.Lock()` held (if this is testable, not sure).

Reply:
Going to leave this as is. Having `s.reading()` panic seems fine, and other State methods don't test for this.